include/capstone/arc.h

```diff
@@ -0,0 +1,375 @@
+#ifndef CAPSTONE_ARC_H
+#define CAPSTONE_ARC_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if !defined(_MSC_VER) || !defined(_KERNEL_MODE)
+#include <stdint.h>
+#endif
+
+#include "platform.h"
+#include "cs_operand.h"
+
+/// Operand type for instruction's operands
+typedef enum arc_op_type {
+	ARC_OP_INVALID = CS_OP_INVALID, ///< Invalid
+	ARC_OP_REG = CS_OP_REG,	      ///< Register operand
+	ARC_OP_IMM = CS_OP_IMM,	      ///< Immediate operand
+} arc_op_type;
+
+// /// Instruction's operand referring to memory
+// /// This is associated with ARC_OP_MEM operand type above
+// typedef struct arc_op_mem {
+// 	unsigned int base;  ///< base register
+// 	unsigned int index; ///< index register
+// 	int64_t disp;	    ///< displacement/offset value
+// } arc_op_mem;
+
+/// Instruction operand
+typedef struct cs_arc_op {
+	arc_op_type type;	      //< operand type
+	union {
+		unsigned int reg;     /// register value for REG operand
+		int64_t imm;	      /// immediate value for IMM operand
+		// arc_op_mem mem; /// base/disp value for MEM operand
```

Comment:
Do you want to add the memory operand in the future? If so we should do it before the v6 Gold release. Because changing the API afterwards is meh.

Reply:
There's a problem here: not all instructions that actually use MemRI go through `printMemOperandRI`. As a result, it turns out that, for example, the `ldb %r0, [%r1,4]` instruction will contain operand.mem, while `ldh.ab %r0, [%r1,4]` will not, despite they are similar

Reply:
I see. Please open an issue about it. So we can track it.